src/som/interpreter/SArguments.java

```diff
@@ -35,7 +37,15 @@ public static DynamicObject getEnvironment(final Frame frame) {
   }
   
   public static ExecutionLevel getExecutionLevel(final Frame frame) {
-    return (ExecutionLevel)args(frame)[EXECUTION_LEVEL_IDX];
+    Object execLevel = args(frame)[EXECUTION_LEVEL_IDX];
+    if (execLevel.getClass() == ExecutionLevel.class) {
+      // Graal can't know that only this branch is ever taken, so, we need to help it a little
+      return (ExecutionLevel) execLevel;
+    } else {
+      // this should never happen
+      CompilerDirectives.transferToInterpreter();
+      return (ExecutionLevel) execLevel;
+    }
   }
 
```

Comment:
Because this path is never taken.
A branch profile implements the necessary logic so that it still works if the branch is actually executed. But we do not need to support this case.

Generally, the logic with a branch profile is: if this branch has never been taken before, cut it out of compilation. if it has been taken don't do anything. The transfer to interpreter is always 'cut it out of compilation'.